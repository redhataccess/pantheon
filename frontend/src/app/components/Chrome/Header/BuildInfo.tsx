import React, { Component } from 'react';
import { Link } from "react-router-dom";

class BuildInfo extends Component {
    public state = {
        buildDate: '',
        commitHash: '',
        commitText: ''
    };

    public componentDidMount() {
        this.getBuildInfo()
    }

    public render() {
        return (
            <React.Fragment>
                <div className="column-view">
                    {!this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && <Link to=""
                        onClick={this.commitRedirect}>
                        {this.state.buildDate}
                    </Link>}
                    {this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && this.state.buildDate}
                </div>
            </React.Fragment>
        );
    }

    public getBuildInfo() {
        const backend = "/pantheon/builddate.json"
        if (this.state.buildDate === '') {
            fetch(backend)
                .then(response => response.json())
                .then(responseJSON => this.setState({ buildDate: "Build Date: " + responseJSON.buildDate, commitHash: responseJSON.commitHash }))
        }
    };

    private commitRedirect = () => {
        window.location.href = "https://github.com/redhataccess/pantheon/commit/" + this.state.commitHash
    }
}

export { BuildInfo }