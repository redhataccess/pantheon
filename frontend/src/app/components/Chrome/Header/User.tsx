import React, { Component } from 'react'
import { Link } from "react-router-dom"
import { IAppState } from '@app/app'

class User extends Component<IAppState, any> {
    public render() {
        return (
            <React.Fragment>
                <Link to={this.props.userAuthenticated ? '' : '/login'}
                    onClick={this.conditionalRedirect}>
                    {this.props.userAuthenticated ? 'Log Out [' + this.props.username + ']' : 'Log In'}
                </Link>
            </React.Fragment>
        );
    }

    private conditionalRedirect = () => {
        if (this.props.userAuthenticated) {
            fetch('/system/sling/logout')
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }