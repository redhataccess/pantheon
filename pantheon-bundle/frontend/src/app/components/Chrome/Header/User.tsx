import React, { Component } from "react"
import { Link } from "react-router-dom"
import {
    Button, Dropdown,
    DropdownToggle,
    DropdownItem, Modal, DropdownPosition
} from "@patternfly/react-core"
import { HelpIcon } from "@patternfly/react-icons"
import "@app/app.css"
import { IAppState } from "@app/app"

interface IState {
    helpDropdownOpen: boolean
    loginUrl: string
    logoutUrl: string
}

class User extends Component<IAppState, IState> {
    constructor(props) {
        super(props)
        this.state = {
            helpDropdownOpen: false,
            loginUrl: "/auth/login",
            logoutUrl: "/system/sling/logout"
        }
    }

    public render() {
        const dropdownItems = [
            <DropdownItem key="userguide" href="/pantheon/docs/assemblies/assembly-pantheon-help.html" target="_blank">User Guide</DropdownItem>,
            <DropdownItem key="contribute" href="https://github.com/redhataccess/pantheon" target="_blank">Contribute to Pantheon</DropdownItem>
        ]
        return (
            <React.Fragment>
                <Dropdown onSelect={this.onHelpSelect}
                    toggle={
                        <DropdownToggle toggleIndicator={null} onToggle={this.onHelpToggle}>
                            <HelpIcon />
                        </DropdownToggle>
                    }
                    isPlain={true}
                    isOpen={this.state.helpDropdownOpen}
                    dropdownItems={dropdownItems}
                    position={DropdownPosition.right}
                />
                <a className="p2-header__login"
                    href={this.props.userAuthenticated ? this.state.logoutUrl : this.state.loginUrl}
                    onClick={this.conditionalRedirect}>
                    {this.props.userAuthenticated ? "[" + this.props.username + "]" : "Log In"}
                </a>
            </React.Fragment>
        )
    }

    private onHelpToggle = helpDropdownOpen => {
        this.setState({ helpDropdownOpen })
    }

    private onHelpSelect = () => {
        this.setState({
            helpDropdownOpen: !this.state.helpDropdownOpen
        })
    }

    private conditionalRedirect = () => {
        if (this.props.userAuthenticated) {
            fetch(this.state.logoutUrl)
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }
