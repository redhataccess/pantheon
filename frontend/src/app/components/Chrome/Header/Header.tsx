import React from 'react'
import { PageHeader } from '@patternfly/react-core'
import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand />}
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}