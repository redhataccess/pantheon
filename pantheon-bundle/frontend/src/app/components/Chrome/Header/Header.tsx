import React from 'react'
import { PageHeader, Brand } from '@patternfly/react-core'
// import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'
import logo from '../../../images/Pantheon2-logo-white.png';

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand src={logo} alt="Pantheon" className="PageHeader__Brand" />}
            logoProps={
                {
                'href': '/',
                // @todo Replace in CSS file when a project stylesheet has been setup
                'style': {'maxWidth': '220px'}
                }
            }
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}