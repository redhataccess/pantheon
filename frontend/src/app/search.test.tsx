import React from 'react';
import Search from '@app/search';
import { BuildInfo } from './components/Chrome/Header/BuildInfo';
import "isomorphic-fetch"
import { HashRouter as Router } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import { DataList, Button, TextInput } from '@patternfly/react-core';
import renderer from 'react-test-renderer';
import sinon from "sinon";

describe('Search tests', () => {
  test('should render default Search component', () => {
    const view = shallow(<Search />);
    expect(view).toMatchSnapshot();
  });

  it('should render a Button', () => {
    const wrapper = mount(<Router><Search /></Router>);
    const button = wrapper.find(Button);
    expect(button.exists()).toBe(true)
  });

  it('should render a DataList', () => {
    const wrapper = mount(<Router><Search /></Router>);
    const dataList = wrapper.find(DataList);
    expect(dataList.exists()).toBe(true)
  });

  it('should render a TextInput', () => {
    const wrapper = mount(<Router><Search /></Router>);
    const textInput = wrapper.find(TextInput);
    expect(textInput.exists()).toBe(true)
  });

  it('should render a BuildInfo component', () => {
    const wrapper = mount(<Router><Search /></Router>);
    const buildInfo = wrapper.find(BuildInfo);
    expect(buildInfo.exists()).toBe(true)
  });

  it('test click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button onClick={mockCallBack}>Submit</Button>));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('test getRows function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.getRows).toMatchSnapshot();
  });

  it('test buildSearchUrl function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.buildSearchUrl).toMatchSnapshot();
  });

  it('test hideAlertOne function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.hideAlertOne).toMatchSnapshot();
  });

  it('test confirmDeleteOperation function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.confirmDeleteOperation).toMatchSnapshot();
  });

  it('test cancelDeleteOperation function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.cancelDeleteOperation).toMatchSnapshot();
  });

  it('test sortByUploadTime function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.sortByUploadTime).toMatchSnapshot();
  });

  it('test sortByDescription function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.sortByDescription).toMatchSnapshot();
  });

  it('test sortByName function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.sortByName).toMatchSnapshot();
  });

  it('test dismissNotification function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.dismissNotification).toMatchSnapshot();
  });

  it('test doSearch function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.doSearch).toMatchSnapshot();
  });

  it('test newSearch function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.newSearch).toMatchSnapshot();
  });

  it('test delete function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.delete).toMatchSnapshot();
  });

  it('test handleDeleteCheckboxChange function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.handleDeleteCheckboxChange).toMatchSnapshot();
  });

  it('test handleSelectAll function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.handleSelectAll).toMatchSnapshot();
  });

  it('test setInput function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.setInput).toMatchSnapshot();
  });

  test('newSearch() click event', () => {
    const wrapper = shallow(<Search />);
    const instance = wrapper.instance();
    const spy = sinon.spy(instance, 'newSearch');

    wrapper.setState({ "iput": "test" })
    wrapper.find(Button).simulate('click');
    sinon.assert.called(spy);
  });

  it('test fetchTimeout function', () => {
    const wrapper = renderer.create(<Router><Search /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.fetchTimeout).toMatchSnapshot();
  });
});
