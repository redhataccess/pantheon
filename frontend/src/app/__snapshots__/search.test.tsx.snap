// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should render default Search component 1`] = `
<Fragment>
  <div>
    <div>
      <FormGroup
        fieldId="search"
        helperText="Search is case sensitive. An empty search will show all modules."
        label="Search Query"
      >
        <div
          className="row-view"
        >
          <TextInput
            aria-label={null}
            className=""
            id="search"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
          <Component
            onClick={[Function]}
          >
            Search
          </Component>
        </div>
      </FormGroup>
      <div
        className="notification-container"
      >
        <Pagination
          bottom={false}
          handleMoveLeft={[Function]}
          handleMoveRight={[Function]}
          handleMoveToFirst={[Function]}
          handlePerPageLimit={[Function]}
          nextPageRecordCount={1}
          pageNumber={1}
          perPageLimit={25}
          showDropdownOptions={true}
        />
      </div>
      <DataList
        aria-label="Simple data list"
      >
        <DataListItem
          aria-labelledby="simple-item1"
        >
          <DataListItemRow
            id="data-rows-header"
          >
            <DataListCheck
              aria-label="controlled checkbox example"
              aria-labelledby="width-ex1-check1"
              checked={false}
              className="checkbox"
              id="check"
              isChecked={false}
              isDisabled={false}
              onClick={[Function]}
            />
            <DataListItemCells
              dataListCells={
                Array [
                  <DataListCell
                    width={2}
                  >
                    <button
                      aria-label="sort column by name"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Name
                    </button>
                  </DataListCell>,
                  <DataListCell
                    width={2}
                  >
                    <button
                      aria-label="sort column by description"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Description
                    </button>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type"
                    >
                      Source Type
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-name"
                    >
                      Source Name
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <button
                      aria-label="sort column by upload time"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Upload Time
                    </button>
                  </DataListCell>,
                ]
              }
            />
          </DataListItemRow>
          <DataListItemRow
            id="data-rows"
          />
          <DataListItemRow
            id="data-rows"
          >
            <DataListCheck
              aria-label="controlled checkbox example"
              aria-labelledby="width-ex3-check1"
              checked={false}
              className="checkbox"
              id=""
              isChecked={false}
              name=""
              onClick={[Function]}
            />
            <DataListItemCells
              dataListCells={
                Array [
                  <DataListCell
                    width={2}
                  >
                    <Link
                      to=""
                    >
                      
                    </Link>
                  </DataListCell>,
                  <DataListCell
                    width={2}
                  >
                    <span>
                      
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span />
                  </DataListCell>,
                  <DataListCell>
                    <span>
                      
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span>
                      
                    </span>
                  </DataListCell>,
                ]
              }
              key=""
            />
          </DataListItemRow>
          <DataListItemRow
            id="data-rows"
          />
        </DataListItem>
      </DataList>
      <div
        className="notification-container"
      >
        <Pagination
          bottom={true}
          handleMoveLeft={[Function]}
          handleMoveRight={[Function]}
          handleMoveToFirst={[Function]}
          handlePerPageLimit={[Function]}
          nextPageRecordCount={1}
          pageNumber={1}
          perPageLimit={25}
          showDropdownOptions={false}
        />
        <BuildInfo />
      </div>
      <div
        className="alert"
      />
    </div>
  </div>
</Fragment>
`;

exports[`Search tests test buildSearchUrl function 1`] = `undefined`;

exports[`Search tests test cancelDeleteOperation function 1`] = `undefined`;

exports[`Search tests test confirmDeleteOperation function 1`] = `undefined`;

exports[`Search tests test delete function 1`] = `undefined`;

exports[`Search tests test dismissNotification function 1`] = `undefined`;

exports[`Search tests test doSearch function 1`] = `undefined`;

exports[`Search tests test fetchTimeout function 1`] = `undefined`;

exports[`Search tests test getRows function 1`] = `undefined`;

exports[`Search tests test handleDeleteCheckboxChange function 1`] = `undefined`;

exports[`Search tests test handleSelectAll function 1`] = `undefined`;

exports[`Search tests test hideAlertOne function 1`] = `undefined`;

exports[`Search tests test newSearch function 1`] = `undefined`;

exports[`Search tests test setInput function 1`] = `undefined`;

exports[`Search tests test sortByDescription function 1`] = `undefined`;

exports[`Search tests test sortByName function 1`] = `undefined`;

exports[`Search tests test sortByUploadTime function 1`] = `undefined`;
